llama-cli/src/cli_args.rs

```diff
@@ -1,34 +1,94 @@
 use std::path::PathBuf;
 
-use clap::Parser;
-use llama_rs::TokenBias;
-use once_cell::sync::Lazy;
+use clap::{Parser, ValueEnum};
+use llama_rs::{
+    InferenceParameters, InferenceSessionParameters, ModelKVMemoryType, TokenBias, EOT_TOKEN_ID,
+};
+use rand::SeedableRng;
 
 #[derive(Parser, Debug)]
 #[command(author, version, about, long_about = None)]
-pub struct Args {
-    /// Where to load the model path from
-    #[arg(long, short = 'm')]
-    pub model_path: String,
+pub enum Args {
+    #[command()]
+    /// Use a model to infer the next tokens in a sequence, and exit
+    Infer(Box<Infer>),
+
+    #[command()]
+    /// Use a model to dump the tokens for a given prompt
+    DumpTokens(Box<DumpTokens>),
```

Comment:
The description for "dump tokens is very ambiguous. I'd bring back the original:

> /// Dumps the prompt to console and exits, first as a comma-separated list of token IDs
    /// and then as a list of comma-separated string keys and token ID values.